app/services/file_service.py

```
@@ -174,7 +229,4 @@ def download_from_url(url: str) -> tuple[str, str]:
 
         except requests.RequestException as e:
             logger.error("Failed to download file from URL %s: %s", url, str(e))
-            raise HTTPException(
-                status_code=400,
-                detail=f"Failed to download file from URL: {str(e)}",
-            )
+            raise FileDownloadError(url=url, original_error=e)
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)